src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParamOrderCheck.java

```diff
@@ -0,0 +1,361 @@
+ *   &#47;**
+ *    * No violation for extra "extra" parameter.
+ *    *
+ *    * &#64;version Some javadoc.
+ *    * &#64;param &lt;T&gt; description
+ *    * &#64;param E desc          // violation, wrong order for "E" tag
+ *    * &#64;param &lt;S&gt; desc
+ *    * &#64;param array desc
+ *    * &#64;see Some javadoc      // OK
+ *    * &#64;param e desc          // violation, wrong order for "e" tag
+ *    * &#64;param str desc        // violation, wrong order for "str" tag
+ *    * &#64;param extra desc      // OK, Extra parameter
+ *    * &#64;return Some number
+ *    *&#47;
+ *   public static &lt;T, E, S&gt; int sampleMethod(T[] array, String str, E e ) {
+ *     return 2021;
+ *   }
+ *
+ *   &#47;**
+ *    * No violation for extra "t" parameter.
+ *    *
+ *    * &#64;param t desc          // OK, Extra parameter
+ *    *&#47;
+ *   public void sampleMethod2() {}
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+  * <li>
+ * {@code javadoc.missed.html.close}
+ * </li>
+ * <li>
+ * {@code javadoc.parse.rule.error}
+ * </li>
+ * <li>
+ * {@code javadoc.tag.order}
+ * </li>
+ * <li>
+ * {@code javadoc.wrong.singleton.html.tag}
+ * </li>
+ * </ul>
+ *
+ * @since 8.43
+ */
+@StatelessCheck
+public class JavadocParamOrderCheck extends AbstractJavadocCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_JAVADOC_TAG_ORDER = "javadoc.tag.order";
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.JAVADOC,
+        };
+    }
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return getAcceptableJavadocTokens();
+    }
+
+    /**
+     * Specify the list.
+     */
+    private final List<Integer> targetParent = Arrays.asList(
+        TokenTypes.INTERFACE_DEF,
+        TokenTypes.RECORD_DEF,
+        TokenTypes.MODIFIERS,
+        TokenTypes.TYPE,
+        TokenTypes.TYPE_PARAMETERS,
+        TokenTypes.CTOR_DEF
+    );
+
+    /**
+     * Specify the list.
+     */
+    private final List<Integer> targetSibiling = Arrays.asList(
+        TokenTypes.PARAMETERS,
+        TokenTypes.RECORD_COMPONENTS,
+        TokenTypes.INTERFACE_DEF
+    );
+
+    @Override
+    public void visitJavadocToken(DetailNode node) {
+        DetailAST ast = getBlockCommentAst().getParent();
+
+        if (ast != null) {
+            if (ast.getType() == TokenTypes.ANNOTATION) {
+                ast = ast.getParent();
+            }
+            if (targetParent.contains(ast.getType())) {
+                DetailAST typeDef = ast.getParent();
+                final DetailAST prevSib = getBlockCommentAst().getPreviousSibling();
+                if (prevSib != null && prevSib.getType() == TokenTypes.MODIFIERS) {
+                    typeDef = ast;
+                }
```

Comment:
CI will fail, Branch coverage issue here in this `if`
if anyone finds the soln let me know meanwhile i will also find this and search more and better tests cases.